manifests/package.pp

```diff
@@ -12,9 +12,18 @@
     }
   } else {
     package { $forge_server::package:
-      ensure   => present,
+      ensure   => $::forge_server::pkg_ensure,
       provider => $::forge_server::provider
     }
   }
 
+  # If we are ensuring the latest gem package, uninstall old versions
+  if $::forge_server::pkg_ensure == 'latest' {
```

Comment:
I don't see much value in coming up with complex logic for the cleanup. To me puppet is about converging to a defined state regardless of what the previous state was. And in this class the most important is that forge server gem is ensured anything else is either out of scope or less important.

That being said I'm sure we can find a place for some cleanup code if you have more or less consistent use case, even though I wouldn't advise spending too much time on it.